lib/graphql_devise/rails/routes.rb

```diff
@@ -1,111 +1,79 @@
-        raise GraphqlDevise::Error, 'Trying to skip a non supported operation. Check for typos.'
-      end
-      unless only_operations.all? { |only| supported_operations.include?(only) }
-        raise GraphqlDevise::Error, 'One of the `only` operations is not supported. Check for typos.'
-      end
 
-      path         = opts.fetch(:at, '/graphql_auth')
-      mapping_name = resource.underscore.tr('/', '_').to_sym
+      validate_gql_devise_operations!(param_operations)
 
       devise_for(
         resource.pluralize.underscore.tr('/', '_').to_sym,
         module: :devise,
-        skip:   [:sessions, :registrations, :passwords, :confirmations, :omniauth_callbacks, :unlocks, :invitations]
+        skip:   DEVISE_OPERATIONS
       )
 
-      authenticatable_type = opts[:authenticatable_type] ||
-                             "Types::#{resource}Type".safe_constantize ||
-                             GraphqlDevise::Types::AuthenticatableType
-
-      used_mutations = if only_operations.present?
-        default_mutations.slice(*only_operations)
-      else
-        default_mutations.except(*skipped_operations)
-      end
-      used_mutations.each do |action, mutation|
-        used_mutation = if custom_operations[action].present?
-          custom_operations[action]
-        else
-          new_mutation = Class.new(mutation)
-          new_mutation.graphql_name("#{resource}#{action.to_s.camelize(:upper)}")
-          new_mutation.field(:authenticatable, authenticatable_type, null: true)
+      prepared_mutations = GraphqlDevise::MutationsPreparer.call(
+        resource:             resource,
+        mutations:            GraphqlDevise::OperationSanitizer.call(
+          default: GraphqlDevise::MutationsPreparer::DEFAULT_MUTATIONS, **param_operations
+        ),
+        authenticatable_type: authenticatable_type
+      )
 
-          new_mutation
-        end
-        used_mutation.instance_variable_set(:@resource_name, mapping_name)
+      prepared_queries = GraphqlDevise::QueriesPreparer.call(
+        resource:             resource,
+        queries:              GraphqlDevise::OperationSanitizer.call(
+          default: GraphqlDevise::QueriesPreparer::DEFAULT_QUERIES, **param_operations
+        ),
+        authenticatable_type: authenticatable_type
+      )
 
-        GraphqlDevise::Types::MutationType.field("#{mapping_name}_#{action}", mutation: used_mutation)
-      end
-      additional_mutations.each do |action, mutation|
+      prepared_mutations.merge(additional_mutations).each do |action, mutation|
         GraphqlDevise::Types::MutationType.field(action, mutation: mutation)
       end
-
-      if (used_mutations.present? || additional_mutations.present?) &&
-         (Gem::Version.new(GraphQL::VERSION) <= Gem::Version.new('1.10.0') || GraphqlDevise::Schema.mutation.nil?)
-        GraphqlDevise::Schema.mutation(GraphqlDevise::Types::MutationType)
```

Comment:
There was a reason for this not no be added if the mutation type has no mutations. You will get an error when making a request saying that the mutation type needs at least one field